### DISCUSSION TITLE
Maximum straight-line speed for the Lego Spike Hub, when used with a medium-sized motor, is limited.

### DISCUSSION THREAD

**Discussion by milad-asrar** [pybricks/support#1650] [General]
I am using this code to set / get the straight_speed of a drive base. It is possible to set the straight_speed up to 488 correctly, but
any value about that is limited to 488, eg when I set the speed to 800, I still get 488 with this code.

```python
from pybricks.hubs import PrimeHub
from pybricks.pupdevices import Motor, ColorSensor, UltrasonicSensor, ForceSensor
from pybricks.parameters import Button, Color, Direction, Port, Side, Stop, Axis
from pybricks.robotics import DriveBase
   
hub = PrimeHub(top_side=Axis.Z, front_side=Axis.X)  
motor_l = Motor(Port.A, Direction.COUNTERCLOCKWISE)
motor_r = Motor(Port.E) 
db = DriveBase(motor_l, motor_r, wheel_diameter=56, axle_track=200)
drive_speed = 800
db.settings(straight_speed=drive_speed) 
print(db.settings())
```
Result: (488, 733, 93, 420)

**Comment by nothingtodowu** [comment]
I am also new to Pybricks, I try to explain this. db.ctl_steps_per_app_step = motor.ctl_steps_per_app_step (1000) * ROT_MDEG_OVER_PI (114592) / wheel_diameter (56000) = 2046

drive_speed = speed_max(1000000) / db.ctl_steps_per_app_step (2046) = 488.

So if you use wheel_diameter = 56 mm then your speed limit is 488. 

BTW, if you change to large wheel, wheel_diameter = 87, then you can get higher speed 765.

**Reply by panthertrader** [reply]
thank you

**Comment by laurensvalk** [comment]
Coincidentally, the reason that these caps exist was illustrated quite well just now in https://github.com/orgs/pybricks/discussions/1649 

Raising the speed limit might seem like a good idea at first, but pushing beyond what the motors can practically do means that there isn't any wiggle room for the motors to correct for errors. Arguably, perhaps the limits should be lower, not higher :-)

**Reply by nothingtodowu** [reply]
Hi, then how about use gear? Gear should not effect correct errors. I think motor with gear just like use a bigger wheel. Can I do like that to increase speed?

**Reply by laurensvalk** [reply]
> Gear should not effect correct errors.

Gears usually introduce backlash, meaning that the wheels can move around more even while the motor isn't moving.

Bigger wheels don't have this problem, but that only works if it fits in your design.

Generally though, driving faster also means being a bit less precise, so be careful about making it too fast. Slow and steady wins the race :)

**Reply by nothingtodowu** [reply]
Agree! Precise is more important to competition.
